### PR TITLE
MM-45713 - change 500 error to json object

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This PR changes the invalid email response from being a 5xx error to be a forbidden and returning a json object that contains the string "true" for valid business emails and "false" for invalid ones.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45713

#### Related Pull Requests
https://github.com/mattermost/mattermost-webapp/pull/10800

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: api4/cloud.go [hunk 1]
```diff
@@ -232,12 +232,17 @@ func validateBusinessEmail(c *Context, w http.ResponseWriter, r *http.Request) {
...
 
-	errValidatingEmail := c.App.Cloud().ValidateBusinessEmail(user.Id, emailToValidate.Email)
-	if errValidatingEmail != nil {
-		c.Err = model.NewAppError("Api4.valiateBusinessEmail", "api.cloud.request_error", nil, errValidatingEmail.Error(), http.StatusInternalServerError)
+	emailErr := c.App.Cloud().ValidateBusinessEmail(user.Id, emailToValidate.Email)
+	if emailErr != nil {
+		c.Err = model.NewAppError("Api4.validateBusinessEmail", "api.cloud.request_error", nil, emailErr.Error(), http.StatusForbidden)
+		emailResp := model.ValidateBusinessEmailResponse{IsValid: "false"}
+		json, _ := json.Marshal(emailResp)
+		w.Write(json)
```
[the remark this answers is not in the document]
Comment (pvev) [reply api4/cloud.go:240]: ok, I will! thanks for the feedback @agnivade 👍

**Review by agnivade — APPROVED** [review]

File: model/cloud.go [hunk 2]
```diff
@@ -105,6 +105,10 @@ type ValidateBusinessEmailRequest struct {
 	Email string `json:"email"`
 }
 
+type ValidateBusinessEmailResponse struct {
+	IsValid string `json:"is_valid"`
```
Comment (JulienTant) [inline model/cloud.go:109]: nit: I don't see a reason to not use a `bool` here, we always want to return `true` or `false`.

Overall I believe the more accurate the type, the better quality of life for maintenance etc.

**Review by JulienTant — APPROVED** [review]
LGTM, just a nit over using "string" as "bool"

**Review by noxer — APPROVED** [review]

File: model/cloud.go (hunk 2 above)
Comment (neallred) [reply model/cloud.go:109]: ditto it would be better to represent as a bool. It would simplify the code in the frontend as well.

**Review by neallred — APPROVED** [review]
LGTM, thanks @pvev !

**Pull request merged**